docker/dockerfile_build.bzl

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule for building a Docker image from a Dockerfile and saving it
+   into a tar file.
+
+This rule uses the docker tool installed in the system to build the image and
+then save it as a tar file. The saved tar file is available for other rules to
+use (e.g. container_image, container_load, container_test)
+
+The produced tar file is available at @<repo_name>//image:dockerfile_image.tar
+"""
+
+def docker(repository_ctx):
+    """Resolves the docker path.
+
+    Args:
+      repository_ctx: The repository context
+
+    Returns:
+      Tha path to the docker tool
+    """
+
+    if repository_ctx.attr.docker_path:
+        return repository_ctx.attr.docker_path
+    if repository_ctx.which("docker"):
+        return repository_ctx.which("docker")
+
+    fail("Path to the docker tool was not provided and it could not be resolved automatically.")
+
+def _impl(repository_ctx):
+    """Core implementation of dockerfile_image."""
+
+    docker_path = docker(repository_ctx)
+
+    dockerfile_src = repository_ctx.path(repository_ctx.attr.dockerfile)
+    dockerfile_dest = repository_ctx.path("Dockerfile")
+
+    # Copy the provided Dockerfile into the root of this workspace since using
+    # the -f flag with the docker build command is problematic.
+    cp_result = repository_ctx.execute([
+        "cp",
+        dockerfile_src,
+        dockerfile_dest,
+    ])
+    if cp_result.return_code:
+        fail("Failed to copy Dockerfile from {} to {}: {}".format(
+            dockerfile_src,
+            dockerfile_dest,
+            cp_result.stderr,
+        ))
+
+    img_name = repository_ctx.name + ":dockerfile_image"
+    build_args = [
+        docker_path,
+        "build",
+        "-t",
+        img_name,
+        repository_ctx.path(""),
+    ]
+    build_result = repository_ctx.execute(build_args)
+    if build_result.return_code:
+        fail("docker build command failed: {} ({})".format(
+            build_result.stderr,
+            " ".join(build_args),
+        ))
+
+    image_tar = "dockerfile_image.tar"
```

Comment:
document somewhere the image tar will be produced with this name

Reply:
This was documented in the top level comment.